exp/api/v1alpha3/types.go

```diff
@@ -0,0 +1,45 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha3
+
+import (
+	infrav1 "sigs.k8s.io/cluster-api-provider-azure/api/v1alpha3"
+)
+
+type (
+	VMSSVM struct {
+		ID               string          `json:"id,omitempty"`
+		InstanceID       string          `json:"instanceID,omitempty"`
+		Name             string          `json:"name,omitempty"`
+		AvailabilityZone string          `json:"availabilityZone,omitempty"`
+		State            infrav1.VMState `json:"vmState,omitempty"`
+	}
+
+	VMSS struct {
+		ID            string             `json:"id,omitempty"`
+		Name          string             `json:"name,omitempty"`
+		Sku           string             `json:"sku,omitempty"`
+		Capacity      int64              `json:"capacity,omitempty"`
+		Zones         []string           `json:"zones,omitempty"`
+		Image         infrav1.Image      `json:"image,omitempty"`
+		StartupScript string             `json:"startupScript,omitempty"`
```

Comment:
Does this map to custom data? Even though "CustomData" is confusing, aliasing might cause more confusion.

Reply:
Agreed. Again, following the VM structure to keep uniformity.

Perhaps, we should change this in a refactor PR?

Reply:
I don't think that field is used

Reply:
If I'm right and it's not in use, I vote for removing it (for machine pool only) and opening an issue to remove it for VMs. Let's not introduce a new field without using it, it's harder to deprecate API types later.